test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetLocalsCommandTest.cs

```diff
@@ -90,5 +94,88 @@ public void LocalsCommand_Success_ValidLocalResource_ListMessage(string args)
+                                                                                             l.LogInformation == mockConsole.Object.LogInformation &&
+                                                                                             SettingsUtility.GetConfigValue(l.Settings, "foo", false, false) == SettingsUtility.GetConfigValue(defaultSettings, "foo", false, false) &&
+                                                                                             SettingsUtility.GetConfigValue(l.Settings, "kung foo", false, false) == SettingsUtility.GetConfigValue(defaultSettings, "kung foo", false, false))));
+            }
+        }
+
+        [Theory]
+        [InlineData("all")]
+        [InlineData("http-cache")]
+        [InlineData("global-packages")]
+        [InlineData("temp")]
+        public async void LocalsCommand_ParsingValidation_WithConfigParam(string arg)
+        {
+            // Use a test directory to validate test key-value pairs within ISettings object passed to Runner
+            using (var mockCurrentDirectory = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange
+                var dummyConfigPath = Util.CreateDummyConfigFile(mockCurrentDirectory.Path);
+                var mockLocalsCommandRunner = new Mock<ILocalsCommandRunner>();
+                var mockConsole = new Mock<IConsole>();
+                mockConsole.Setup(c => c.Verbosity).Returns(Verbosity.Detailed);
+                LocalsCommand localsCommand = new LocalsCommand();
+                localsCommand.LocalsCommandRunner = mockLocalsCommandRunner.Object;
+                localsCommand.Clear = true;
+                localsCommand.List = false;
+                localsCommand.Console = mockConsole.Object;
+                localsCommand.Arguments.Add(arg);
+                localsCommand.ConfigFile = dummyConfigPath;
+                localsCommand.CurrentDirectory = mockCurrentDirectory.Path;
+                var directory = Path.GetDirectoryName(dummyConfigPath);
+                var configFileName = Path.GetFileName(dummyConfigPath);
+                var defaultSettings = Configuration.Settings.LoadDefaultSettings(directory,
+                                                                                 configFileName,
+                                                                                 localsCommand.MachineWideSettings);
+
+                // Act
+                localsCommand.Execute();
+                await localsCommand.ExecuteCommandAsync();
+
+                // Assert
+                mockLocalsCommandRunner.Verify(mock => mock.ExecuteCommand(It.Is<LocalsArgs>(l => !l.List && l.Clear &&
+                                                                                             l.Arguments.Count == 1 && l.Arguments[0] == arg &&
+                                                                                             l.LogError == mockConsole.Object.LogError &&
```

Comment:
`ReferenceEquals` instead of `==` shows the intent here a bit better, IMO. Also, I don't see you `Setup` the `LogError` and `LogInformation` projects, so they are just null, right?